AtSheet/explainer.md

```diff
@@ -0,0 +1,225 @@
+
+## Proposal - `@sheet`
+Create a new `@sheet` CSS block, for separating style sheets with named identifiers.
+
+All examples use a stylesheet cased as `sheet.css` with the following contents:
+```css
+div {
+  color: blue;
+}
+
+@sheet foo {
+  div {
+    color: red;
+  }
+}
+@sheet bar {
+  div {
+    font-family: sans-serif;
+  }
+}
+```
+
+## Proposal - Importing a specific sheet via `@import`
+```html
+<style>
+  @import sheet("sheet.css#foo");
+</style>
+```
+
+This will import only this rules for `foo` - in this case, the `div { color: red; }` rule, and will *not* import any rules from `sheet.css` outside of "foo".
+
+## Proposal - Importing a specific sheet via the `<link>` tag
+```html
+<link rel="stylesheet" href="sheet.css#foo" />
+```
+
+This will also import only this rules for "foo" - in this case, the `div { color: red; }` rule, and will *not* import any rules from `sheet.css` outside of "foo".
+
+## Proposal - Importing a base set of inline styles into a Declarative Shadow DOM
+Shadow DOM isolates styles, but fragment identifiers are global. This enables Declarative Shadow DOM to import `@sheet` references from the light DOM.
+
+```html
+<style>
+@sheet foo {
+  div {
+    color: red;
+  }
+}
+</style>
+<template shadowrootmode="open">
+  <link rel="stylesheet" href="#foo" />
+  <span>I'm in the shadow DOM</span>
+</template>
+```
+or imported from JS:
+```html
+<script>
+  import {foo} from './sheet.css' with {type: 'css'};
+  ...
+  shadow.adoptedStyleSheets = [foo];
+</script>
+```
+
+## Detailed design discussion
+
+#### Named Imports with Imperative Shadow DOM
+
+`sheet.js` can also be imported via Javascript as follows:
+
+```js
+import baz, { bar } from 'sheet.css' with { type: 'css' }
+```
+
+`baz` will reference style rules outside of any `@sheet` blocks as a Default Import (in this case, the `div { color: blue; } ` rule).
+
+`bar` will reference style rules within the `@sheet bar` block as a Named Import (in this case, the `div { color: red; }  ` rule).
+
+Named imports may be renamed as part of this import process:
+
+```js
+import baz, { bar as renamed } from 'sheet.css' with { type: 'css' }
+```
+
+The default import may be omitted, importing only the named `@sheet`:
+
+```js
+import { bar } from 'sheet.css' with { type: 'css' }
+```
+
+Any of these `import` examples can be then used to set the `adoptedStyleSheets` attribute on a Shadow DOM node:
+
+```js
+import { bar } from 'sheet.css' with { type: 'css' }
+document.adoptedStyleSheets = [bar];
+shadowRoot.adoptedStyleSheets = [bar];
+```
+
+#### Performance
+
+This will be a performance-neutral feature, and use of it may allow for developers to reduce the number of network requests. We should ensure that multiple imports of different sheets from the same file produce a single network request. 
```

Comment:
"and use of it may allow for developers" -> "but developers can utilize this feature to reduce..."